server/src/CompilationDatabase.cpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * \file
+ *
+ * \brief Using libclang for reading the JSON compilation database.
+ *
+ * This file is distributed under the GNU General Public License. See
+ * COPYING for details.
+ */
+#include <iostream>
+
+#include <clang-c/Index.h>
+#include <clang-c/CXCompilationDatabase.h>
+#include <clang-c/CXString.h>
+
+#include "CompilationDatabase.h"
+
+static void getExactCompCmds(CXCompilationDatabase &db,
+                             const std::string &fullFilename,
+                             std::vector<std::vector<std::string>> &flags) {
+  CXCompileCommands cmds =
+      clang_CompilationDatabase_getCompileCommands(db, fullFilename.c_str());
+  unsigned ncmd = clang_CompileCommands_getSize(cmds);
+  flags.resize(ncmd);
+
+  for (unsigned i = 0; i < ncmd; ++i) {
+    CXCompileCommand cmd = clang_CompileCommands_getCommand(cmds, 0);
```

Comment:
Should be `i`?

Don't correct, I'm writting you a patch that I will submit back to you so you can take a look.